### PR TITLE
fix: #976 fixed checkbox in ReportsPlans

### LINKED ISSUE #935: User's ways: job breaks down when string contains link markdown used as job name
Maybe similar to [729](https://github.com/tritonJS826/masters-way/issues/729)

## Description

Job breaks down when string contains link markdown with empty name part used as job name

## Preconditions

1. Created at least one Way
2. Created at least one Day report in Way

## Steps to reproduce

1. Create new job
2. Enter link markdown with empty name part, e.g. `[](https://google.com)`
3. Click on any part of page for save entered data

## Actual result

The line collapses. You cannot enter a job name.

## <a id="expected-result">Expected result</a>

Spaces are trimmed and a placeholder is displayed 
image

**or** a visually empty line is shown

**or** automatically converted to custom placeholder
image

### LINKED ISSUE #976: Day Reports: Button “Cancel” and the cross in the dialog box that occurs when clicking on the check box in Plans block, doesn’t work correctly
Preconditions
User is authorized. User's way is created and opened. Some day reports with plans exist 

Steps
1.	Click on the check box next to any written plan in Plans block to mark it as completed
2.	Click on “Cancel” button in the dialog box
3.	Repeat with clicking on the cross 

Expected result
Plan isn’t marked as completed

Actual result
Plan is marked as completed

https://github.com/tritonJS826/masters-way/assets/163824877/83824494-9803-4f26-bcc7-586010cee697

https://github.com/tritonJS826/masters-way/assets/163824877/b73cd1c0-c1b0-47d2-b8ab-4262108c5c71

### PR DESCRIPTION
**Pull request by AKiroi**
- fixed #976
 - fixed #935
 - global styles

### CODE REVIEW

File: mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx [hunk 1]
```diff
@@ -312,8 +312,10 @@ export const ReportsTablePlansCell = observer((props: ReportsTablePlansCellProps
                   <Modal
                     trigger={
                       <Checkbox
+                        isDisabled={true}
                         isDefaultChecked={plan.isDone}
-                        onChange={() => {}}
+                        // OnChange={() => {}}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx:317]:
```suggestion
```

File: mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx [hunk 2]
```diff
@@ -312,8 +312,9 @@ export const ReportsTablePlansCell = observer((props: ReportsTablePlansCellProps
                   <Modal
                     trigger={
                       <Checkbox
+                        isDisabled={true}
                         isDefaultChecked={plan.isDone}
-                        onChange={() => {}}
+                        onChange={(isDone) => isDone}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx:317]: waht does this function do?

File: mw-webapp/src/utils/markdown/renderMarkdown.tsx [hunk 3]
```diff
@@ -16,7 +20,7 @@ const customComponents: Components = {
         rel="noopener noreferrer"
         className={styles.a}
       >
-        {children}
+        {!children ? LanguageService.common.emptyLinkTitle[language] : children}
```
Comment (tritonJS826) [inline mw-webapp/src/utils/markdown/renderMarkdown.tsx:23]: children ?? LanguageService.common.emptyLinkTitle[language]

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx (hunk 2 above)
Comment (tritonJS826) [reply mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx:317]: if you copy it from somewhere - check those place too

File: mw-webapp/src/utils/markdown/renderMarkdown.tsx (hunk 3 above)
Comment (AKiroi) [reply mw-webapp/src/utils/markdown/renderMarkdown.tsx:23]: fixed

File: mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx (hunk 1 above)
Comment (AKiroi) [reply mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx]: fixed

**Review by Ekaterina1994 — APPROVED** [review]

File: mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx [hunk 4]
```diff
@@ -312,6 +312,7 @@ export const ReportsTablePlansCell = observer((props: ReportsTablePlansCellProps
                   <Modal
                     trigger={
                       <Checkbox
+                        isDisabled={true}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx:315]: Should it be always disabled?

Comment (AKiroi) [reply mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx:315]: Yes, by default isDisabled set to false and the checkbox triggers (changes its value) every time it's clicked. However, if we explicitly set it to true, we can control the click on it

Comment (AKiroi) [reply mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/reportsTablePlansCell/ReportsTablePlansCell.tsx:315]: We disable fuction handleCheckboxChange
```
  const handleCheckboxChange = () => {
    setIsChecked(!isChecked);
    props.onChange(!isChecked);
  };
```
and the checkbox will only trigger when props.isDefaultChecked changes

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**